sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -757,6 +768,93 @@ Exceptions:
 * Throws synchronously with error code `invalid` if the resulting dependency would
   lead to a cycle. This error is omitted when `property::graph::no_cycle_check` is set.
 
+|
+[source,c++]
+----
+node add_barrier(const std::vector<event> eventList = {});
+----
+
+|Adds a barrier to the graph.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::modifiable`.
+
+Parameters:
+
+* `eventList` - Zero or more events to wait for.
+  These events can be graph-limited events (from graph recording) 
+  or general sycl events (such as events external to the graph).
+  If no event are provided, the barrier waits for all the previous nodes 
+  of the graph to complete.
+
+Returns: The command-group function object node which has been added to the graph.
+
+|
+[source,c++]
+----
+node make_external_event(event& recordedEvent, event& externalEvent, bool updatable = false);
+----
+
+|Turns a graph-limited event into an external event on which other operations 
+can wait (including other graphs).
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::modifiable`.
+
+Parameters:
+
+* `recordedEvent` - Graph-limited event that will be linked to 
+  the external event.
+
+* `externalEvent` - Sycl event to be used as external event.
+
+* `updatable` - Flag indicating that users will update the external event 
+  after the graph finalization. Please, note that setting this flag may degrade 
+  graph execution performance for level-zero backend, as updatable events 
+  are handled by the host.
+
+Returns: The command-group function object node which has been added to the graph.
```

Comment:
In theory it could be a possible use case to also pass this node as a dependency for another node (if you wanted to explicitly block a kernel on this event being signalled or something like that). In practice though I don't know if that's worth supporting.

A node here would also support queries for dependencies and things like that (and could have a query for getting the associated external event) which are potentially useful. 

I think on balance I'm in favour of keeping the node for consistency as well as the above.